lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -83,36 +83,38 @@ public ParquetSchemaConverter(List<Type> types, List<String> columnNames, boolea
         checkArgument(types.size() == columnNames.size(), "types size not equals to columnNames size");
         this.useLegacyDecimalEncoding = useLegacyDecimalEncoding;
         this.useInt96TimestampEncoding = useInt96TimestampEncoding;
-        this.messageType = convert(types, columnNames);
+        ImmutableMap.Builder<List<String>, Type> primitiveTypesBuilder = ImmutableMap.builder();
+        messageType = convert(types, columnNames, primitiveTypesBuilder);
+        primitiveTypes = primitiveTypesBuilder.buildOrThrow();
     }
 
-    private MessageType convert(List<Type> types, List<String> columnNames)
+    private MessageType convert(List<Type> types, List<String> columnNames, ImmutableMap.Builder<List<String>, Type> primitiveTypesBuilder)
```

Comment:
can this be static now?

Reply:
Only if we pass the boolean values as well

Reply:
Makes sense because we don't really need the boolean fields anyway.